scheduler/plugin/guest.go

```diff
@@ -67,24 +68,45 @@ func (pl *wasmPlugin) newGuest(ctx context.Context) (*guest, error) {
 		out.Reset()
 	}
 
+	// Allocate a call stack sized to max of params / return values of any
+	// guest function.
+	callStack := make([]uint64, 1)
+
 	return &guest{
 		guest:     g,
 		out:       &out,
 		filterFn:  g.ExportedFunction(guestExportFilter),
-		callStack: make([]uint64, 1), // Sized to max of params / return values of filterFn
+		scoreFn:   g.ExportedFunction(guestExportScore),
+		callStack: callStack,
 	}, nil
 }
 
-// filter calls the WebAssembly guest function handler.FuncHandleRequest.
+// filter calls guestExportFilter.
 func (g *guest) filter(ctx context.Context) *framework.Status {
 	defer g.out.Reset()
 	callStack := g.callStack
+
 	if err := g.filterFn.CallWithStack(ctx, callStack); err != nil {
 		return framework.AsStatus(decorateError(g.out, "filter", err))
 	}
-	code := uint32(callStack[0])
-	reason := filterParamsFromContext(ctx).reason
-	return framework.NewStatus(framework.Code(code), reason)
+	statusCode := int32(callStack[0])
+	statusReason := paramsFromContext(ctx).reason
+	return framework.NewStatus(framework.Code(statusCode), statusReason)
+}
+
+// score calls guestExportScore.
+func (g *guest) score(ctx context.Context) (int64, *framework.Status) {
+	defer g.out.Reset()
+	callStack := g.callStack
+
+	if err := g.scoreFn.CallWithStack(ctx, callStack); err != nil {
+		return 0, framework.AsStatus(decorateError(g.out, "score", err))
+	}
+
+	score := int32(callStack[0] >> 32)
```

Comment:
added tests to make sure this part wasn't goofed (e.g. cast to uint32 was here until we switched to signed)